src/modding/ModConfigItemInfo.cs

```diff
@@ -0,0 +1,214 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using Godot;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+/// <summary>
+///   Class that holds the info of a mod config item from a 'mod_config.json' file
+/// </summary>
+public class ModConfigItemInfo : HBoxContainer
+{
+    public string? ID { get; set; }
+
+    [JsonProperty("Display Name")]
+    public string? DisplayName { get; set; }
+
+    public string? Description { get; set; }
+
+    public string? Type { get; set; }
+
+    // Exclusively for Enums/Option type
+    public object? Options { get; set; }
+
+    public object? Value { get; set; }
```

Comment:
This sounds rather permissive and thus error prone. As I'm not full aware of how modding works as of now, I won't say this isn't the way to go, but I'd like to raise some awareness on it.

PS: Why not going, e.g., for a generic type ?

Reply:
I saw a bit that this seems to add like a value parsing type thing for some reason?
I think for that I'd like to see a more bulletproof system than manually coding some parser stuff...
Perhaps something like using a JSON object defining the settings or something and then code that creates controls based on the types in that or something. I especially was horrified how a bunch of different aliases could refer to the same types in the configuration.

Reply:
I removed the different aliases but i am unsure on how to make this better as the type variable is already like what you are suggesting, I tried making Value a dynamic variable but i ran into compiler errors also i tried using generics but i either don't understand them enough or it wouldn't work as the way i am storing them won't allow for different types of configitems to be stored together

Reply:
It just feels wrong to me to try to write a custom configuration language for Thrive.
If you really want to include a way to support mods having settings available with a system built into Thrive, I think a better approach would be to like either specify it as a JSON structure or as a C# class where the data types can be detected through reflection.

Reply:
So would it be better if the gui(the actual control) and the value type(int, string, bool) were separate variables instead of how it is currently (with stuff like int range vs int)?

Reply:
I'm currently doing a major refactoring of the microbe editor to split it into many different parts. I would have to spend the brain power to actually really design this part entirely to be able to say what's a good way. I can only really say that this kind of custom configuration GUI creating from untyped objects is not something I particularly like. Would you perhaps consider making this part a separate PR so that the non-controversial modding improvements could be merged sooner? Also I worry that this kind of complex mod configuration needs a very in-depth guide for modders to even this system *at all*.

Reply:
Yeah sure, that's probably better